src/app/shared/pipes/filter.pipe.ts

```diff
@@ -0,0 +1,22 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Course } from '../interfaces/course.interface.';
+
+@Pipe({
+  name: 'filter'
+})
+export class FilterPipe implements PipeTransform {
+
+  transform(courses: Course[], searchtext: string): Course[] {
+    if (!courses) {
+      return [];
+    }
+    if (!searchtext) {
+      return courses;
+    }
+    searchtext = searchtext.toLowerCase();
+    return courses.filter(course => {
```

Comment:
Please add the type for the **course**

Reply:
But it was mentioned in the return type of transform method - transform(courses: Course[], searchtext: string): **Course[]**